src/emulator/emulator.cpp

```diff
@@ -356,197 +350,42 @@ long __faasm_read_input(unsigned char* buffer, long bufferLen)
-    logger->debug(
-      "E - chain_this_knative idx {} input length {}", idx, bufferLen);
-
-    const std::string host = faabric::util::getEnvVar("FAASM_INVOKE_HOST", "");
-    const std::string port = faabric::util::getEnvVar("FAASM_INVOKE_PORT", "");
-
-    if (host.empty() || port.empty()) {
-        logger->error(
-          "Expected FAASM_INVOKE_HOST and FAASM_INVOKE_PORT to be set");
-        throw std::runtime_error("Missing invoke host and port");
-    }
-
-    int portInt = std::stoi(port);
-
-    // Build the message to dispatch
-    faabric::Message msg =
-      faabric::util::messageFactory(_emulatedCall.user(), funcName);
-    msg.set_idx(idx);
-    msg.set_inputdata(buffer, bufferLen);
-    msg.set_ispython(_emulatedCall.ispython());
-    msg.set_pythonuser(_emulatedCall.pythonuser());
-    msg.set_pythonfunction(_emulatedCall.pythonfunction());
-
-    if (pyName != nullptr) {
-        msg.set_pythonentry(pyName);
-    }
-
-    // Chained calls are always async and can be awaited by the caller
-    msg.set_isasync(true);
-
-    std::string msgJson = faabric::util::messageToJson(msg);
-    logger->debug("POST {}:{} ({})", host, portInt, msgJson);
-
-    // Flush stdout before chaining
-    fflush(stdout);
-
-    // Make the call
-    faabric::util::postJsonFunctionCall(host, portInt, msg);
-    return msg.id();
-}
-
-unsigned int __faasm_chain_function(const char* name,
-                                    const unsigned char* inputData,
-                                    long inputDataSize)
-{
-    faabric::util::SystemConfig& conf = faabric::util::getSystemConfig();
-    if (conf.hostType == "knative") {
-        return _chain_knative(name, 0, 0, inputData, inputDataSize);
-    } else {
-        throw std::runtime_error("Chaining by name not supported in emulator");
-    }
+    throw std::runtime_error("Chaining not supported in emulator");
 }
 
-unsigned int __faasm_chain_this(int idx,
+unsigned int __faasm_chain_ptr(int (*funcPtr)(),
                                 const unsigned char* buffer,
                                 long bufferLen)
 {
-    faabric::util::SystemConfig& conf = faabric::util::getSystemConfig();
-    if (conf.hostType == "knative") {
-        return _chain_knative(
-          _emulatedCall.function(), idx, nullptr, buffer, bufferLen);
-    } else {
-        return _chain_local(idx, nullptr, buffer, bufferLen);
-    }
+    throw std::runtime_error("Chaining not supported in emulator");
 }
 
 unsigned int __faasm_chain_py(const char* name,
                               const unsigned char* buffer,
                               long bufferLen)
 {
-    faabric::util::SystemConfig& conf = faabric::util::getSystemConfig();
-    if (conf.hostType == "knative") {
-        return _chain_knative(
-          _emulatedCall.function(), 0, name, buffer, bufferLen);
-    } else {
-        return _chain_local(0, name, buffer, bufferLen);
-    }
-}
-
-int _await_call_knative(unsigned int callId)
```

Comment:
This stuff was all about supporting native code execution in Knative which we won't do any more.